### PR TITLE
Avoid potential divide by zero when misaligning

### PR DESCRIPTION
**Pull request by eveenhuis**
Removes possibility for divide by zero if all distances are zero when "misaligning." Also adds some additional error checking

### CODE REVIEW

**Comment by dmjoy** [comment]
LGTM, no objections to merging.  Just a side note though; usually (if we're not dealing with numpy arrays here) I would explicitly check for a zero denominator and set the value to infinity or otherwise handle that condition more explicitly as using an epsilon can go unnoticed or be confusing if it isn't called out.

**Comment by eveenhuis** [comment]
> LGTM, no objections to merging. Just a side note though; usually (if we're not dealing with numpy arrays here) I would explicitly check for a zero denominator and set the value to infinity or otherwise handle that condition more explicitly as using an epsilon can go unnoticed or be confusing if it isn't called out.

Your comment is making me question whether epsilon needs to be added to the numerator or not (and if that even matters). I was thinking that the values are only needed for relative comparison, so it probably (?) makes sense to operate on non-infinite values, but maybe not?

**Comment by jadie1** [comment]
> Edit: I'm suggesting maybe it should just be ` probs = [dist/(sum(dists) + eps) for dist in dists]`

I think this would be sufficient

File: align_system/utils/alignment_utils.py [hunk 1]
```diff
@@ -53,7 +59,9 @@ def _select_min_dist_choice(self, choices, dists, misaligned=False, probabilisti
             # maximize over non-inverted distances
 
             # Convert distances to probabilities
-            probs = [dist/sum(dists) for dist in dists]
+            # Avoid divide by zero if all distances are 0
+            total_dist = sum(dists) + eps * len(dists)
```
Comment (dmjoy) [inline align_system/utils/alignment_utils.py:63]: Does it make more sense here to do `sum(dists) * len(dists) + eps` (we don't need to multiply the epsilon right?)

Comment (dmjoy) [reply align_system/utils/alignment_utils.py:63]: (Or maybe I'm just confused about the intent here)

Comment (eveenhuis) [reply align_system/utils/alignment_utils.py:63]: Yeah, I think we should remove `len(dists)` from the equation entirely `sum(dists) + eps`

Comment (eveenhuis) [reply align_system/utils/alignment_utils.py]: Alright what do we think now? (`probs = [dist/(sum(dists) + eps) for dist in dists]`)

**Pull request merged**